### PR TITLE
Move two-layer shallow water equations from Trixi.jl

### PR DESCRIPTION
**Pull request by patrickersing**
This PR adds the 1D and 2D version of the `ShallowWaterEquationsTwoLayer`, that were removed from `Trixi.jl` in https://github.com/trixi-framework/Trixi.jl/pull/1809

### CODE REVIEW

**Comment by patrickersing** [comment]
This is basically a direct copy from the files in `Trixi.jl` and I only had to qualify some of the functions.
I slightly restructured the testsets, which causes most of the deleted lines in this PR.

**Review by andrewwinters5000 — CHANGES_REQUESTED** [review]
Just a couple questions, but overall this already looks good! It is nice how simple it was to load in a new set of equations.

File: src/equations/shallow_water_two_layer_1d.jl [hunk 1]
```diff
@@ -0,0 +1,507 @@
...
+  <https://www.sciencedirect.com/bookseries/international-geophysics/vol/101/suppl/C>\
+  ISBN: 978-0-12-088759-0
+"""
+struct ShallowWaterTwoLayerEquations1D{RealT <: Real} <:
+       Trixi.AbstractShallowWaterEquations{1, 5}
+    gravity::RealT   # gravitational constant
+    H0::RealT        # constant "lake-at-rest" total water height
+    rho_upper::RealT # lower layer density
+    rho_lower::RealT # upper layer density
+    r::RealT         # ratio of rho_upper / rho_lower
```
[the remark this answers is not in the document]
Comment (patrickersing) [reply src/equations/shallow_water_two_layer_1d.jl:67]: Yes, for now I wanted to just move the existing Trixi.jl  functionalities.
The additional wet/dry treatment and shock capturing for the two-layer equations will then be added in an extra PR.

File: src/equations/equations.jl [hunk 2]
```diff
@@ -10,4 +10,7 @@
 
 include("shallow_water_wet_dry_1d.jl")
 include("shallow_water_wet_dry_2d.jl")
+
```
[the remark this answers is not in the document]
Comment (patrickersing) [reply src/equations/equations.jl:13]: Yes, we definitely should do that once we add some more equations.  I was already thinking about this when I added the equation files, but felt it was a bit superfluous to add additional comments for just these two equations.

[the remark this answers is not in the document]
Comment (andrewwinters5000) [reply src/equations/equations.jl:13]: I agree that it is a bit of overkill now, but we can keep this in mind for the future.

**Review by andrewwinters5000 — APPROVED** [review]
LGTM!

**Pull request merged**